### PR TITLE
Convert many examples to TypeScript

### PR DESCRIPTION
**Pull request by maxproske**
Strategized with @balazsorban44 to open one larger PR, with changes to individual examples as separate commits. 

For each example, I researched how multiple realworld codebases use the featured technology with TypeScript, to thoughtfully convert them by hand - nothing automated whatsoever.

## Documentation / Examples

- [X] Make sure the linting passes by running `pnpm lint`
- [X] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

### CODE REVIEW

**Review by balazsorban44 — DISMISSED** [review]
Impressive work, thanks!

File: examples/environment-variables/package.json [hunk 1]
```diff
@@ -7,7 +7,13 @@
   },
   "dependencies": {
     "next": "latest",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
+  },
+  "devDependencies": {
```
Comment (balazsorban44) [inline examples/environment-variables/package.json:13]: Examples like this should add the `tsconfig.json` file by default. Could you address that?

Comment (maxproske) [reply examples/environment-variables/package.json:13]: @balazsorban44 Sorry I don't follow, there is a `tsconfig.json` file present in all examples.

**Review by balazsorban44 — APPROVED** [review]

File: examples/environment-variables/package.json (hunk 1 above)
Comment (balazsorban44) [reply examples/environment-variables/package.json:13]: You are correct, I was likely looking at the wrong branch. Thanks!

**Pull request merged**